### ISSUE TITLE
Rule S4502 Disabling CSRF protections is security-sensitive

### ISSUE THREAD

**Issue by eric-therond-sonarsource** [SonarSource/SonarJS#1828]
Implements [RSPEC-4502](https://jira.sonarsource.com/browse/RSPEC-4502)

For both JS/TS

We should report on each `app.METHOD` (check full list of methods https://expressjs.com/en/4x/api.html#routing-methods) which is not using `csurf` middleware. We ignore `app.get`, `app.head`, `app.options` as they are always safe.

`csurf` middleware can be provided in 2 ways:
* Globally for the Express.js `app`. In this case it's important that `app.use(csrfProtection)` is called before the `app.METHOD`
```
var app = express();
var csrf = require('csurf');
var csrfProtection = csrf({ cookie: { httpOnly: true, secure:true }})
app.use(csrfProtection);
app.post(....)  // Compliant
```
* Locally for the particular `app.METHOD` call
```
var app = express();
var csrf = require('csurf');
var csrfProtection = csrf({ cookie: { httpOnly: true, secure:true }})
app.post(..., csrfProtection, ...)  // Compliant
```

In both cases `csurf` protection should not be considered effective if any unsafe method (like `POST`) is listed in `ignoreMethods` property. Unsafe methods are all methods except safe (`GET`, `HEAD`, `OPTIONS`)
```
var app = express();
var csrf = require('csurf');
var csrfProtection = csrf({ cookie: { httpOnly: true, secure:true, ignoreMethods: ["POST", "GET"] }}); // `POST` is unsafe, so this protection is not effective
app.post(..., csrfProtection, ...)  // Noncompliant
```

Code examples: https://github.com/SonarSource/security-expected-issues/tree/master/javascript/rules/hotspots/RSPEC-4502
